dotCMS/src/main/java/com/dotmarketing/util/ImportStarterUtil.java

```diff
@@ -0,0 +1,1148 @@
+                FileUtil.copyDirectory(f.getPath(), ad.getPath() + File.separator + f.getName());
+            } else {
+                FileUtil.copyFile(f.getPath(), ad.getPath() + File.separator + f.getName());
+            }
+        }
+    }
+
+    /**
+     * Deletes all files from the backupTempFilePath
+     */
+    private void deleteTempFiles() {
+        File f = new File(backupTempFilePath);
+
+        FileUtil.deltree(f, true);
+    }
+
+
+
+    private void deleteTable(final String table) {
+
+        final DotConnect dotConnect = new DotConnect();
+        dotConnect.setSQL("delete from " + table);
+        dotConnect.getResult();
+    }
+
+    interface ObjectFilter {
+        boolean includeIt(Object obj);
+    }
+
+    /**
+     * This method takes an XML file and will try to import it via XStream and Hibernate.
+     *
+     * @param f - File to be parsed and imported.
+     * @param out - Printwriter to write responses to Reponse Printwriter so this method can write to
+     *        screen.
+     */
+    private void doXMLFileImport(File f) throws Exception {
+        doXMLFileImport(f, null);
+    }
+    
+    
+    
+
+    /**
+     * This method takes an XML file and will try to import it via XStream and Hibernate.
+     *
+     * @param f - File to be parsed and imported.
+     * @param out - Printwriter to write responses to Reponse Printwriter so this method can write to
+     *        screen.
+     * @param filter
+     * @throws DotDataException
+     * @throws HibernateException
+     * @throws EncryptorException
+     * @throws SQLException 
+     * @throws DotSecurityException 
+     * @throws DuplicateUserException 
+     * @throws NoSuchMethodException 
+     * @throws InvocationTargetException 
+     * @throws IllegalAccessException 
+     */
+    private void doXMLFileImport(File f, ObjectFilter filter) throws Exception {
+        if (f == null) {
+            return;
+        }
+
+
+
+            Class _importClass = null;
+            HibernateUtil _dh = null;
+
+            boolean counter = false;
+            boolean image = false;
+            boolean portlet = false;
+            boolean logsMappers = false;
+
+
+            Pattern classNamePattern = Pattern.compile("_[0-9]{8}");
+            Logger.info(this, "**** Importing the file: " + f + " *****");
+
+            /* if we have a multipart import file */
+            
+            final String _className = classNamePattern.matcher(f.getName()).find() 
+                   ?  f.getName().substring(0, f.getName().lastIndexOf("_"))
+                   :   f.getName().substring(0, f.getName().lastIndexOf("."));
+
+
+            if (_className.equals("Counter")) {
+                counter = true;
+            } else if (_className.equals("Image")) {
+                image = true;
+            } else if (_className.equals("Portlet")) {
+                portlet = true;
+            } else if (_className.equals("LogsMappers")) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)